rasa/nlu/classifiers/regex_message_handler.py

```diff
@@ -0,0 +1,283 @@
+                f"Expected the intent to be one of [{domain.intents}] "
+                f"but found {intent_name}."
+                f"Continuing with given line as user text.",
+                docs=DOCS_URL_STORIES,
+            )
+            intent_name = None
+        return intent_name
+
+    @staticmethod
+    def _parse_optional_entities(match: Match, domain: Domain) -> List[Dict[Text, Any]]:
+        """Extracts the optional entity information from the given pattern match.
+
+        If no entities are specified or if the extraction fails, then an empty list
+        is returned.
+
+        Args:
+            match: a match produced by `self.pattern`
+            domain: the domain
+        Returns:
+            some list of entities
+        """
+        entities_str = match.group(ENTITIES)
+        if entities_str is None:
+            return []
+
+        try:
+            parsed_entities = json.loads(entities_str)
+            if not isinstance(parsed_entities, dict):
+                raise ValueError(
+                    f"Parsed value isn't a json object "
+                    f"(instead parser found '{type(parsed_entities)}')"
+                )
+        except (JSONDecodeError, ValueError) as e:
+            rasa.shared.utils.io.raise_warning(
+                f"Failed to parse arguments in line '{match.string}'. "
+                f"Failed to decode parameters as a json object (dict). "
+                f"Make sure the intent is followed by a proper json object (dict). "
+                f"Continuing without entities. "
+                f"Error: {e}",
+                docs=DOCS_URL_STORIES,
+            )
+            parsed_entities = dict()
+
+        # validate the given entity types
+        entity_types = set(parsed_entities.keys())
+        unknown_entity_types = entity_types.difference(domain.entities)
+        if unknown_entity_types:
+            rasa.shared.utils.io.raise_warning(
+                f"Failed to parse arguments in line '{match.string}'. "
+                f"Expected entities from {domain.entities} "
+                f"but found {unknown_entity_types}. "
+                f"Continuing without unknown entity types. ",
+                docs=DOCS_URL_STORIES,
+            )
+            parsed_entities = {
+                key: value
+                for key, value in parsed_entities
+                if key not in unknown_entity_types
+            }
+
+        # convert them into the list of dictionaries that we expect
+        entities: List[Dict[Text, Any]] = []
+        for entity_type, entity_values in parsed_entities.items():
+            if not isinstance(entity_values, list):
+                entity_values = [entity_values]
+            for entity_value in entity_values:
+                entities.append(
+                    {
+                        ENTITY_ATTRIBUTE_TYPE: entity_type,
+                        ENTITY_ATTRIBUTE_VALUE: entity_value,
```

Comment:
mhm, seems the current `RegexInterpreter` doesn't handle entity `role` / `group`. Guess it's fine to omit this from here then as well for now

Reply:
yeah, someone would need to adapt the pattern for that as well

Reply:
(because currently, the pattern is {<type> : <value>} not {"type": type, ...})